rs/bitcoin/kyt/src/main.rs

```diff
@@ -31,6 +25,40 @@ fn check_address(args: CheckAddressArgs) -> CheckAddressResponse {
     }
 }
 
+#[ic_cdk::update]
+/// Return `Passed` if all input addresses of the transaction of the given
+/// transaction id passed the KYT check, or `Failed` if any of them did not.
+///
+/// Every call to check_transaction must attach at least `CHECK_TRANSACTION_CYCLES_REQUIRED`.
+/// Return `NotEnoughCycles` if not enough cycles are attached.
+///
+/// The actual cycle cost may be well less than `CHECK_TRANSACTION_CYCLES_REQUIRED`, and
+/// unspent cycles will be refunded back to the caller, minus a
+/// `CHECK_TRANSACTION_CYCLES_SERVICE_FEE`, which is always deducted regardless.
+///
+/// In certain cases, it may also return `HighLoad` or `Pending` to indicate the
+/// caller needs to call again (with at least `CHECK_TRANSACTION_CYCLES_REQUIRED` cycles)
+/// in order to get the result.
+///
+/// If a permanent error occurred in the process, e.g, when a transaction data
+/// fails to decode or its transaction id does not match, then `Error` is returned
+/// together with a text description.
+async fn check_transaction(args: CheckTransactionArgs) -> CheckTransactionResponse {
+    ic_cdk::api::call::msg_cycles_accept128(CHECK_TRANSACTION_CYCLES_SERVICE_FEE);
+    match Txid::try_from(args.txid.as_ref()) {
+        Ok(txid) => {
+            if ic_cdk::api::call::msg_cycles_available128() + CHECK_TRANSACTION_CYCLES_SERVICE_FEE
+                < CHECK_TRANSACTION_CYCLES_REQUIRED
+            {
+                CheckTransactionResponse::NotEnoughCycles
+            } else {
+                check_transaction_inputs(txid).await
+            }
+        }
+        Err(err) => CheckTransactionResponse::Error(format!("Invalid txid: {}", err)),
```

Comment:
We already return an `Error` case, which can cover this check.

Reply:
> which can cover this check.

Sure, but I think it's more about which errors, from a caller perspective, makes sense. I have the impression it would be ok to panic here, because the caller should not have called us with this input (in the same way we could even fail the call during deserialization of the arguments, before starting any logic). I don't feel strongly about it, either way is fine.

Reply:
Now there is a dedicated CheckTransactionError type with better distinction between different errors, including one for `txid`. I'm inclined not to panic here.